### DISCUSSION TITLE
Deprecating and removing Okta Authentication

### DISCUSSION THREAD

**Discussion by marcolarosa** [Arkisto-Platform/describo-online#48] [Polls]
To the community:

When we first wrote describo we needed a way to authenticate users; after all, you can't have a service online that lets the internet in when it talks to backend storage systems! Since the Uni of Melbourne was using Okta and UTS was moving towards it, we decided to integrate that as our auth service.

Since then, we've come to see Describo as a service that is embedded into another - as an owncloud app for example. And others are embedding it in to their own apps. In that mode, the service describo is embedded into handles the authentication and then just delegates to describo.

So, given this, I'd like to deprecate and remove the Okta auth stack from describo. If it's not in use or planned to be used then it's less code to keep up to date and ensure is secure.

**Comment by beepsoft** [comment]
Hi Marco,

we are trying to implement both authentication and storage based on Dataverse. For implementing the storage interface using the Dataverse API we follow the way the Reva storage is implemented via Reva's API.

For authentication it seems that Describo provides this mechanism, which you described above ("Describo as a service that is embedded into another"):

https://github.com/Arkisto-Platform/describo-online/wiki/Integrating-Describo-Online-into-another-service#configuring-describo-to-allow-another-service-to-setup-a-user-session

However it is not clear for me how is this supposed to be implemented, what is the actual flow of information?

As I understand the "application" (Dataverse in our case) would call https://{describo-online}/api/session/application to share a logged in user's authentication token, which would be used in a "postCrateTo" call.

However in the description I see this:

  "postCrateTo": {
    "url": ["https://{your url}"](https://%7Byoururl%7D/),
    "headers": { "Authorization": "Bearer yyy", "Content-Type": "application/json" }
   }

which suggests that a static Bearer token is to be configured in the configuration.json.

Could you please help me to understand how this flow of authentication and data exchange is supposed to work?

Thanks!

**Reply by marcolarosa** [reply]
Hi @beepsoft 


The first part is telling describo that another application is allowed to talk to it. That's the section described [here](https://github.com/Arkisto-Platform/describo-online/wiki/Integrating-Describo-Online-into-another-service#configuring-describo-to-allow-another-service-to-setup-a-user-session). The `postCrateTo` configuration is if you want describo to post the crate data  somewhere else each time it's updated. I don't know that anyone is actually using that.

After you've configured describo to allow connections from your application, you then need to configure your app to setup the session in describo. That's described [here](https://github.com/Arkisto-Platform/describo-online/wiki/Integrating-Describo-Online-into-another-service#setting-up-the-external-service-to-call-describo) and the rest of that page describes what other information you can set up.

If you setup a session inside describo, then you will get a sessionId back that you can use inside an iframe in `your` app to show describo to the user:

```
If successful, a session identifier will be returned from describo online. Call describo-online 
inside an iframe as https://{describo-online}/application?sid={sessionId}. From that point on the 
session id will be used to identify the currently logged in user bypassing all okta authentication.
```


So the flow is something like:

* [Deploy Describo and set it up to accept session setup commands from your app](https://github.com/Arkisto-Platform/describo-online/wiki/Integrating-Describo-Online-into-another-service#setting-up-the-configuration-for-the-external-service)
* [Your app POST's to describo to start a session](https://github.com/Arkisto-Platform/describo-online/wiki/Integrating-Describo-Online-into-another-service#setting-up-the-external-service-to-call-describo)
* Describo returns a sessionId that you then inside an iframe in your app
